### PR TITLE
Add create index and drop index support

### PR DESCRIPTION
**Pull request by mashuai**
This PR adds support for:

    CREATE [ UNIQUE ] INDEX [ IF NOT EXISTS ]
        <index_name>
        ON <table_name>
        ( col_name [, ...] )

    DROP INDEX <index_name>

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -476,6 +476,15 @@ pub enum Statement {
         file_format: Option<FileFormat>,
         location: Option<String>,
     },
+    /// CREATE INDEX
+    CreateIndex {
+        /// index name
+        name: ObjectName,
+        table_name: ObjectName,
+        columns: Vec<ObjectName>,
+        if_unique: bool,
```
Comment (nickolay) [inline src/ast/mod.rs:485]: Simply `unique` seems more appropriate.

File: src/dialect/keywords.rs [hunk 2]
```diff
@@ -50,6 +50,7 @@ macro_rules! define_keywords {
 }
 
 define_keywords!(
+    INDEX,
```
Comment (nickolay) [inline src/dialect/keywords.rs:53]: This ought to be further below.

File: src/parser.rs [hunk 3]
```diff
@@ -912,8 +916,10 @@ impl Parser {
             ObjectType::Table
         } else if self.parse_keyword("VIEW") {
             ObjectType::View
+        } else if self.parse_keyword("INDEX") {
+            ObjectType::Index
         } else {
-            return self.expected("TABLE or VIEW after DROP", self.peek_token());
+            return self.expected("TABLE ,VIEW or INDEX after DROP", self.peek_token());
```
Comment (nickolay) [inline src/parser.rs:922]: Typo.

**Comment by mashuai** [comment]
@nickolay Thanks for the comment. And yes I should use `self.parse_comma_separated(Parser::parse_identifier)?` to parse column names.
 Actually I don't intend to support trailing comma, just because I find `parse_columns` support it, so I do the same thing.

**Comment by nickolay** [comment]
Alright, supporting `CREATE INDEX idx_name ON test (col_name[, ...])` sounds great to me!

In this case, there's no reason to have `parse_index_column` as a separate method (it's rather short now!), this would lead to changing `let object_names` into `let columns`, as expected by `Statement::CreateIndex`. If you can make this last change and squash the changes into a single commit rebased on top of the current master, I'll merge it now. Or I can take it from here, if you prefer.

**Comment by mashuai** [comment]
Hi @nickolay, thanks for the advice, I have moved `parse_index_column` into `parse_create_index` and squash the changes into a single commit rebased on top of the current master.

**Comment by nickolay** [comment]
Excellent, thanks so much for bearing with me and for taking it to the finish line!

I just realized this can now use `parse_parenthesized_column_list`, but as I wanted to credit you in the CHANGELOG anyway, I've gone ahead and made the last minute change myself.

**Pull request merged**